arith_amd64.s

```diff
@@ -0,0 +1,85 @@
+//go:build amd64 && !purego
+
+#include "textflag.h"
+
+// func andAVX2(z, x, y *[32]byte)
```

Comment:
Please comment a bit more, e.g, "andAVX2 performs z = x & y. It modifies `z` in-place and has no return value".